### PR TITLE
Remove extra call to newHscEnvEqWithImportPaths

### PR DESCRIPTION
**Pull request by nlander**
Submitting this at the request of @wz1000 
Instead of calling `newHscEnvEqWithImportPaths` again in `ghcSessionDepsDefinition`, we only generate a new `Unique` for the `HscEnvEq` and update the `envUnique` field.

### CODE REVIEW

**Comment by nlander** [comment]
We also update the `hscEnv` field.

**Review by fendor — APPROVED** [review]
LGTM!

**Comment by michaelpj** [comment]
Worth a comment explaining what's going on?

**Comment by wz1000** [comment]
`newHscEnvEqWithImportPaths` recreates the `ExportsMap` for package dependencies on every call, we probably only want to do that once per session instead of once per call to `ghcSessionDepsDefinition` (which is called for every single file we need to compile).

**Comment by michaelpj** [comment]
put it in the code!

File: ghcide/src/Development/IDE/Core/Rules.hs [hunk 1]
```diff
@@ -799,7 +799,12 @@ ghcSessionDepsDefinition fullModSummary GhcSessionDepsConfig{..} env file = do
 #endif
             session' <- liftIO $ mergeEnvs hsc moduleNode inLoadOrder depSessions
 
-            Just <$> liftIO (newHscEnvEqWithImportPaths (envImportPaths env) session' [])
+            -- Here we avoid a call to to `newHscEnvEqWithImportPaths`, which creates a new
```
Comment (nlander) [inline ghcide/src/Development/IDE/Core/Rules.hs:802]: @michaelpj Does this sufficiently explain what's going on here? Is there anything we should add?

Comment (michaelpj) [reply ghcide/src/Development/IDE/Core/Rules.hs:802]: lgtm

**Review by michaelpj — APPROVED** [review]

File: ghcide/src/Development/IDE/Core/Rules.hs (hunk 1 above)
Comment (michaelpj) [reply ghcide/src/Development/IDE/Core/Rules.hs:802]: I would say ask yourself:
1. If I was reading this for the first time, would I know what is going on and why?
2. Has this captured the knowledge that I gained in order to make the fix?

**Pull request merged**